pykwasm/src/pykwasm/kdist/wasm-semantics/wasm.md

```diff
@@ -520,6 +520,11 @@ Note that, unlike in the WebAssembly specification document, we do not need the
  // ---------------------------------------------------------------------
     rule <instrs> #br_table(ES) => #br(#getInts(ES, minInt(VAL, #lenInts(ES) -Int 1))) ... </instrs>
          <valstack> <i32> VAL : VALSTACK => VALSTACK </valstack>
+      requires 0 <=Int VAL
+      [preserves-definedness]
+      // preserves-definedness:
+      // - 0 <= VAL and minInt(VAL, #lenInts(ES) -Int 1) ensures #getInts(_) is defined
```

Comment:
Is `ES` guaranteed to be non-empty here? Otherwise we'd need a `#lenInts(ES) >Int 0`, too.

Reply:
It is somehow guaranteed by an implementation choice in our semantics. In the official semantics this instruction has two fields: `#br_table(LIST, DEFAULT)`. If the index is not in bounds, `DEFAULT` value is returned. In our syntax, all the arguments are parsed as a single list so the list is non-empty if the wasm code is valid.

Anyway, since non-emptiness is not ensured by the data structure, it would be better if we add the side condition.